### PR TITLE
Model Layer version 3, now with TypeScript, MobX

### PR DESCRIPTION
**Pull request by kring**
This is a work in progress, not at all ready to be merged or even used. I'm just opening it now for a place to jot down some notes.

### CODE REVIEW

File: lib/Models/Cesium.ts [hunk 1]
```diff
@@ -0,0 +1,1221 @@
...
+const createImageryLayer: (ip: Cesium.ImageryProvider) => Cesium.ImageryLayer = createTransformer((ip: Cesium.ImageryProvider) => {
+    console.log('Creating a new ImageryLayer');
+    return new ImageryLayer(ip);
+});
+
+function makeImageryLayerFromParts(parts: ImageryParts): Cesium.ImageryLayer {
+    const layer = createImageryLayer(parts.imageryProvider);
+
+    // This should only be called on ImageryParts that are observable. I think we can assume
+    autorun(() => {
```
Comment (kring) [inline lib/Models/Cesium.ts]: I don't think this autorun is needed; we can just assign the properties once here. If the alpha changes, this function will automatically be invoked again and update the Cesium layer's `alpha` property.

Comment (kring) [reply lib/Models/Cesium.ts]: If I'm right, that's a really good thing because currently this autorun is a leak.

Comment (tephenavies) [reply lib/Models/Cesium.ts]: That'll run the entire map items <-> Cesium ImageryLayerCollection every time you change opacity

Comment (kring) [reply lib/Models/Cesium.ts]: As discussed on Slack, yeah, that mapping re-runs every time, but it's not as expensive as it sounds, and it runs every time even with this autorun in place. Might be room for improvement, but I think it's pretty much fine as is.

**Comment by tephenavies** [comment]
Closing as the comments here are no longer relevant, the `mobx` branch is old and we will open a new PR from `next` to `master` when we are ready.

**Pull request closed without merging**